README.md

```diff
@@ -782,6 +782,21 @@ let printListWithOffset a list1 =
             printfn "%d" (a + elem)
         )
         list1
+
+// OK
+let printListWithOffset a list1 =
+    list1
+    |> List.iter (fun elem ->
+        printfn "%d" (a + elem)
+    )
+
+// OK
```

Comment:
Hi @Smaug123, what if I have something like

```fsharp
    innerFunc<'a, 'b>
        path'
        elementNameThatHasThisRatherLongVariableNameToForceTheWholeThingOnMultipleLines
        (foo >> bar value >> List.item a)
        shape
```

should this be
```fsharp
    innerFunc<'a, 'b> path' elementNameThatHasThisRatherLongVariableNameToForceTheWholeThingOnMultipleLines (
        foo >> bar value >> List.item a
    ) shape
```

Or is there a stricter rule in this case?

Reply:
I think the whole thing wants to split then, i.e. your first example:

```fsharp
    innerFunc<'a, 'b>
        path'
        elementNameThatHasThisRatherLongVariableNameToForceTheWholeThingOnMultipleLines
        (foo >> bar value >> List.item a)
        shape
```

Reply:
So when is it

A:
```fsharp
let printListWithOffset a list1 =
    list1
    |> List.iter (
        ((+) a)
        >> printfn "%d"
    )
```

and when is it

B:
```fsharp
 innerFunc<'a, 'b>
        path'
        elementNameThatHasThisRatherLongVariableNameToForceTheWholeThingOnMultipleLines
        (foo >> bar value >> List.item a)
        shape
```

I'm still sort of looking for a technical rule so to speak.
A could be because:
- No additional arguments are passed to `List.iter`
- A pipe operator is preceding `List.iter`

B could be because:
- Everything before `(foo >> bar value >> List.item a)` already makes the whole thing too long.
- There are additional arguments next to `(foo >> bar value >> List.item a)`, in A there are not.

Reply:
I think it's due to the fact that the lambda is the only argument (ignoring the one we are getting piped in).

For instance:
```
let foo =
    List.iter (
        ((+) a)
        >> printfn "%d"
    )
```

Would be allowed whereas:

```
let foo =
    someOtherThing a b c d (
       ((+) a)
       >> printfn "%d"
    ) f g h
```

Would be reformatted like B

Reply:
I agree: if there is just one argument that is too long, the whole thing gets split over many lines.

Reply:
To be clear, the A example is a bit misleading. These are both right:

```fsharp
let printListWithOffset a list1 =
    list1
    |> List.iter (
        ((+) veryVeryVeryVeryVeryVeryVeryVeryVeryLongThing)
        >> printfn "%d"
    )

let printListWithOffset' a list1 =
    list1
    |> List.iter (((+) a) >> printfn "%d")
```